tensorflow_probability/python/experimental/stats/sample_stats.py

```diff
@@ -266,6 +271,111 @@ def __init__(self, shape=(), dtype=tf.float32):
       shape: Python `Tuple` or `TensorShape` representing the shape of
         incoming samples. By default, the shape is assumed to be scalar.
       dtype: Dtype of incoming samples and the resulting statistics.
-        By default, the dtype is `tf.float32`.
+        By default, the dtype is `tf.float32`. Any integer dtypes will also
+        be treated as `tf.float32` (to not lose significant precision).
     """
     super(RunningVariance, self).__init__(shape, event_ndims=0, dtype=dtype)
+
+
+RunningMeanState = collections.namedtuple(
+    'RunningMeanState', 'num_samples, mean')
+
+
+class RunningMean(object):
+  """Holds metadata for and computes a running mean.
+
+  In computation, samples can be provided individually or in chunks. A
+  "chunk" of size M implies incorporating M samples into a single expectation
+  computation at once, which is more efficient than one by one. If more than one
+  callable is accepted and chunking is enabled, the chunked `axis` will define
+  chunking semantics for all callables.
+
+  `RunningMean` objects do not hold state information. That information,
+  which includes intermediate calculations, are held in a
+  `RunningMeanState` as returned via `initialize` and `update` method
+  calls.
+
+  `RunningMean` is meant to serve general streaming expectations.
+  For a specialized version that fits streaming over MCMC samples, see
+  `ExpectationsReducer` in `tfp.experimental.mcmc`.
+  """
+
+  def __init__(self, shape, dtype=tf.float32):
+    """Instantiates this object.
+
+    Args:
+      shape: Python `Tuple` or `TensorShape` representing the shape of
+        incoming samples.
+      dtype: Dtype of incoming samples and the resulting statistics.
+        By default, the dtype is `tf.float32`. Any integer dtypes will also
+        be treated as `tf.float32` (to not lose significant precision).
```

Comment:
This comment isn't quite accurate.  A 32-bit float has only 23 bits of precision for representing integers, whereas a 32-bit int has 31.  And, of course, an int64 has 63 bits of precision for representing integers.  What you meant was that we need floating-point Tensors because the running mean should be doing floating-point rather than integer division, so say that.  You might also consider a more thoughtful casting strategy that uses 64-bit floating-point Tensors for running means on 64-bit integers.